### ISSUE TITLE
Custom YAML policy logic not working correctly

### ISSUE THREAD

**Issue by d47zm3** [bridgecrewio/checkov#6736]
**Describe the issue**
I'm trying to create custom check in YAML, that would check if non-DLQ AWS SQS queues contain redrive policy, however seems that my check if not working properly and DLQ queues are being reported as non-compliant. What am I doing wrong?

**Examples**
Check in YAML
```yaml
---
metadata:
  name: "Check All Non-DLQ SQS Queues Have Redrive Policy"
  id: "CKV2_CUSTOM_AWS_SQS_REDRIVE_POLICY"
  category: "GENERAL_SECURITY"
definition:
  and:
    - cond_type: "attribute"
      attribute: "name"
      operator: "not_ending_with"
      value: "-dlq"
    - cond_type: "attribute"
      resource_types:
        - aws_sqs_queue
      attribute: "redrive_policy"
      operator: "exists"
```

Terraform
```hcl
resource "aws_sqs_queue" "sample_dlq" {
  name                      = "sample-dlq"
  tags                      = local.tags
  message_retention_seconds = var.sqs_retention_period
}

resource "aws_sqs_queue" "sample" {
  name                       = "sample"
  tags                       = local.tags
  visibility_timeout_seconds = 60
  message_retention_seconds  = var.sqs_retention_period

  redrive_policy = jsonencode({
    deadLetterTargetArn = aws_sqs_queue.sample_dlq.arn
    maxReceiveCount     = 1
  })
}
```

Execution
```
checkov -f sqs2.tf --external-checks-dir ./checks -c CKV2_CUSTOM_AWS_SQS_REDRIVE_POLICY
```

Output
```
By Prisma Cloud | version: 3.2.255

terraform scan results:

Passed checks: 1, Failed checks: 1, Skipped checks: 0

Check: CKV2_CUSTOM_AWS_SQS_REDRIVE_POLICY: "Check All Non-DLQ SQS Queues Have Redrive Policy"
        PASSED for resource: aws_sqs_queue.sample
        File: /sqs2.tf:7-17
Check: CKV2_CUSTOM_AWS_SQS_REDRIVE_POLICY: "Check All Non-DLQ SQS Queues Have Redrive Policy"
        FAILED for resource: aws_sqs_queue.sample_dlq
        File: /sqs2.tf:1-5

                1 | resource "aws_sqs_queue" "sample_dlq" {
                2 |   name                      = "sample-dlq"
                3 |   tags                      = local.tags
                4 |   message_retention_seconds = var.sqs_retention_period
                5 | }
```

**Version (please complete the following information):**
3.2.255

**Additional context**
